vizro-core/tests/integration/test_navigation.py

```diff
@@ -8,6 +8,7 @@
 @pytest.fixture
 def dashboard_result(request):
     # Inject the navigation into the dashboard. Note we need to call request.param since they are all lambda functions.
+    Vizro._reset()
```

Comment:
I don't 100% understand why this needs to be called at the beginning instead of where the current `model_manager._clear(`) call is, so not sure if this is the correct solution @antonymilne, but this seems to fix the failing integration tests.

 The solution in the comments also does not work, so yielding the dashboard and then calling `Vizro._reset()` after. Somehow this only works if `Vizro._reset()` is called before the dashboard definition. But i am wondering why this didn't have interdependence issues before..

Reply:
Yeah, this is correct how you have it. It's definitely not quite right overall but I don't think we'll easily be able to get it any better than this because I spent ages before trying to improve it and couldn't. I'll just updated and moved the comment in [f6b3279](https://github.com/mckinsey/vizro/pull/366/commits/f6b3279537f1be7442374173fefe3d7a0c63b472) but good work figuring this out 👍